python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -0,0 +1,265 @@
+        Return type depends on input:
+        - list-like: DatetimeIndex
+        - Series: Series of datetime64 dtype
+        - scalar: Timestamp
+
+    Examples
+    --------
+    Assembling a datetime from multiple columns of a DataFrame. The keys can be
+    common abbreviations like ['year', 'month', 'day', 'minute', 'second',
+    'ms', 'us', 'ns']) or plurals of the same
+    >>> df = cudf.DataFrame({'year': [2015, 2016],
+    ...                    'month': [2, 3],
+    ...                    'day': [4, 5]})
+    >>> cudf.to_datetime(df)
+    0   2015-02-04
+    1   2016-03-05
+    dtype: datetime64[ns]
+    >>> cudf.to_datetime(1490195805, unit='s')
+    numpy.datetime64('2017-03-22T15:16:45.000000000')
+    >>> cudf.to_datetime(1490195805433502912, unit='ns')
+    numpy.datetime64('1780-11-20T01:02:30.494253056')
+    """
+    if arg is None:
+        return None
+
+    if exact is False:
+        raise NotImplementedError("exact support is not yet implemented")
+
+    if origin != "unix":
+        raise NotImplementedError("origin support is not yet implemented")
+
+    if yearfirst:
+        raise NotImplementedError("yearfirst support is not yet implemented")
+
+    try:
+        if isinstance(arg, cudf.DataFrame):
+            # we require at least Ymd
+            required = ["year", "month", "day"]
+            req = sorted(list(set(required) - set(arg._data.names)))
+            if len(req):
+                raise ValueError(
+                    "to assemble mappings requires at least that "
+                    "[year, month, day] be specified: [{required}] "
+                    "is missing".format(required=",".join(req))
+                )
+
+                # replace passed unit with _unit_map
+
+            def get_units(value):
+                if value in _unit_map:
+                    return _unit_map[value]
+
+                # m is case significant
+                if value.lower() in _unit_map:
+                    return _unit_map[value.lower()]
+
+                return value
+
+            unit = {k: get_units(k) for k in arg._data.names}
+            unit_rev = {v: k for k, v in unit.items()}
+
+            # keys we don't recognize
+            excess = sorted(set(unit_rev.keys()) - set(_unit_map.values()))
+            if len(excess):
+                excess = ",".join(excess)
+                raise ValueError(
+                    f"extra keys have been passed to the \
+                        datetime assemblage: [{excess}]"
+                )
+
+            new_series = (
+                arg[unit_rev["year"]].astype("str")
+                + "-"
+                + arg[unit_rev["month"]].astype("str").str.zfill(2)
+                + "-"
+                + arg[unit_rev["day"]].astype("str").str.zfill(2)
+            )
+            format = "%Y-%m-%d"
+            col = new_series._column.as_datetime_column(
+                "datetime64[ns]", format=format
+            )
+
+            time_delta_col = None
```

Comment:
Can we rename this variable? This isn't a time_delta, it's just a time identifier.